hoodie-common/src/main/java/com/uber/hoodie/configs/AbstractJobConfig.java

```diff
@@ -0,0 +1,151 @@
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+import com.uber.hoodie.exception.InvalidJobConfigException;
+import java.io.Serializable;
+import java.lang.reflect.Field;
+import java.util.Map;
+
+public abstract class AbstractJobConfig implements Serializable {
+
+  private static final String SPACE = " ";
+
+  @Parameter(names = {"--help", "-h"}, help = true)
+  public Boolean help = false;
+
+  public String[] getJobConfigsAsCommandOption(String prefix) {
+    return getJobConfigsAsCommandOption(prefix, null);
+  }
+
+  public String[] getJobConfigsAsCommandOption(String prefix, String suffix) {
+    StringBuilder result = new StringBuilder();
+    Field[] fields = this.getClass().getDeclaredFields();
+
+    if (prefix != null) {
+      result.append(prefix);
+      result.append(SPACE);
+    }
+
+    for (Field field : fields) {
+      try {
+        if (field.isSynthetic()) {
+          continue;
+        }
+        Object value = field.get(this);
+        if (value != null) {
+          result.append(field.getAnnotation(com.beust.jcommander.Parameter.class).names()[0]);
+          result.append(SPACE);
+          result.append(value);
+          result.append(SPACE);
+        }
+      } catch (Throwable e) {
+        throw new InvalidJobConfigException("Failed to convert job config to command string ", e);
+      }
+    }
+
+    if (suffix != null) {
+      result.append(suffix);
+    }
+
+    return result.toString().trim().split(SPACE);
+  }
+
+  @Override
+  public String toString() {
+    StringBuilder result = new StringBuilder();
+    Field[] fields = this.getClass().getDeclaredFields();
+
+    result.append(this.getClass().getSimpleName());
+    result.append('{');
+    for (Field field : fields) {
+      if (field.isSynthetic()) {
+        continue;
+      }
+      try {
+        Object value = field.get(this);
+        result.append(field.getName());
+        result.append("='");
+        result.append(value);
+        result.append("', ");
+      }  catch (Throwable e) {
+        throw new InvalidJobConfigException("Failed in execution of 'toString()' " + e.getMessage());
+      }
+    }
+    result.append("help=");
+    result.append(help);
+    result.append('}');
+    return result.toString();
+  }
+
+  public boolean parseJobConfig(String[] args) {
+    return parseJobConfig(args, false);
+  }
+
+  public boolean parseJobConfig(String[] args, boolean showHelpIfArgsLengthZero) {
+    JCommander cmd = JCommander.newBuilder().addObject(this).build();
+    return parse(cmd, args, showHelpIfArgsLengthZero, false);
+  }
+
+  public boolean parseJobConfig(String[] args, boolean showHelpIfArgsLengthZero, boolean exitJobOnHelp) {
+    JCommander cmd = JCommander.newBuilder().addObject(this).build();
+    return parse(cmd, args, showHelpIfArgsLengthZero, exitJobOnHelp);
+  }
+
+  public boolean parseJobConfig(String[] args, Map<String, AbstractJobConfig> commandConfigMap) {
```

Comment:
This looks like a static utility method that should be outside this class

Reply:
Shouldn't be static. Changes done.